contracts/validators/CredentialAtomicQueryV3Validator.sol

```diff
@@ -0,0 +1,226 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.16;
+
+import {CredentialAtomicQueryValidator} from "./CredentialAtomicQueryValidator.sol";
+import {IVerifier} from "../interfaces/IVerifier.sol";
+import {PrimitiveTypeUtils} from "../lib/PrimitiveTypeUtils.sol";
+
+/**
+ * @dev CredentialAtomicQueryV3 validator
+ */
+contract CredentialAtomicQueryV3Validator is CredentialAtomicQueryValidator {
+    struct CredentialAtomicQueryV3 {
+        uint256 schema;
+        uint256 claimPathKey;
+        uint256 operator;
+        uint256 slotIndex;
+        uint256[] value;
+        uint256 queryHash;
+        uint256[] allowedIssuers;
+        string[] circuitIds;
+        bool skipClaimRevocationCheck;
+        // 0 for inclusion in merklized credentials, 1 for non-inclusion and for non-merklized credentials
+        uint256 claimPathNotExists;
+        uint256 linkSessionID;
+        uint256 nullifierSessionID;
+        uint256 proofType;
+        uint256 verifierID;
+        uint256 authEnabled;
```

Comment:
let's remove it from the query.  Because the query will be valid both for Ethereum and regular identities. 
If auth enabled in public signals -> checkGistRoot. Otherwise -> check msg sender and challenge. Use it in checkAuthEnabled